field/composite.go

```diff
@@ -117,55 +119,75 @@ func (f *Composite) SetData(data interface{}) error {
 
 // Pack deserialises data held by the receiver (via SetData)
 // into bytes and returns an error on failure.
-func (f *Composite) Pack() ([]byte, error) {
-	packed, err := f.pack()
+func (f *Composite) WriteTo(w io.Writer) (n int, err error) {
```

Comment:
Is `Pack()` not a useful method to offer? It looks like the initial design was allocating byte slices and now you're trying to offer a streaming/buffered interface? 

Did we make this change because of potential performance problems with finite byte slices (e.g. higher memory allocations, delays populating the entire array over multiple packets)

Reply:
While the reason to change Unpack is properly read from network (when you may not know how much data to read)! Pack was changed for consistency. I could change only Pack in the Message and keep the rest unchanged.

Reply:
Pack -> WriteTo makes sense. Network writes are going to use `io.Writer`. 

We could keep Pack() and use `bytes.Buffer` to maintain backwords compatibility.